lib/snapshot/BUILD.gn

```diff
@@ -256,7 +256,6 @@ compile_platform("strong_platform") {
   is_runtime_mode_release =
       flutter_runtime_mode == "release" || flutter_runtime_mode == "jit_release"
   args = [
-    "--enable-experiment=non-nullable",
```

Comment:
@jonahwilliams - adding `--sound-null-safety` fails here, but otherwise I think this works?  I'm hoping you'll know if this is safe or not.

Reply:
(same comment for the similar Fuchsia related targets for dart_runner and flutter)

Reply:
Fuchsia currently does not have anything in the engine repo that compiles dart code. All of our tests are c++ unit tests and the targets are leftovers from when we moved out of topaz. We are in the process of adding some dart tests and there are some WIP cls that remove the old targets. I'll make sure that those get moved to sound null safety when they are added.

Reply:
This is building the platform snapshot, not a test.

Reply:
fails how?

Reply:
Like this:

```
FAILED: clang_x64/flutter_runner_patched_sdk/platform_strong.dill clang_x64/flutter_runner_patched_sdk/vm_outline_strong.dill 
python ../../third_party/dart/build/gn_run_binary.py compiled_action /b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/dart -Dsdk_hash=d2025958e3 --packages=/b/s/w/ir/cache/builder/src/third_party/dart/.packages --dfe=/b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/snapshots/kernel-service.dart.snapshot /b/s/w/ir/cache/builder/src/third_party/dart/pkg/front_end/tool/_fasta/compile_platform.dart --sound-null-safety --nnbd-agnostic --target=flutter_runner dart:core --single-root-scheme=org-dartlang-sdk --single-root-base=/b/s/w/ir/cache/builder/src/ org-dartlang-sdk:///flutter/shell/platform/fuchsia/flutter/kernel/libraries.json clang_x64/vm_outline_strong.dill clang_x64/flutter_runner_patched_sdk/platform_strong.dill clang_x64/flutter_runner_patched_sdk/vm_outline_strong.dill
Command failed: /b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/dart -Dsdk_hash=d2025958e3 --packages=/b/s/w/ir/cache/builder/src/third_party/dart/.packages --dfe=/b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/snapshots/kernel-service.dart.snapshot /b/s/w/ir/cache/builder/src/third_party/dart/pkg/front_end/tool/_fasta/compile_platform.dart --sound-null-safety --nnbd-agnostic --target=flutter_runner dart:core --single-root-scheme=org-dartlang-sdk --single-root-base=/b/s/w/ir/cache/builder/src/ org-dartlang-sdk:///flutter/shell/platform/fuchsia/flutter/kernel/libraries.json clang_x64/vm_outline_strong.dill clang_x64/flutter_runner_patched_sdk/platform_strong.dill clang_x64/flutter_runner_patched_sdk/vm_outline_strong.dill
output: Usage: compile_platform [options] dart-library-uri libraries.json vm_outline_strong.dill platform.dill outline.dill

Compiles Dart SDK platform to the Dill/Kernel IR format.

Frequently used options:

  -o <file> Generate the output into <file>.
  -h        Display this message (add -v for information about all options).
Error: Unknown option '--sound-null-safety'.

[6735/7165] ACTION //flutter/shell/platform/fuchsia/flutter/kernel:kernel_platform_files(//build/toolchain/fuchsia:fuchsia)
FAILED: flutter_runner_patched_sdk/platform_strong.dill flutter_runner_patched_sdk/vm_outline_strong.dill 
python ../../third_party/dart/build/gn_run_binary.py compiled_action /b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/dart -Dsdk_hash=d2025958e3 --packages=/b/s/w/ir/cache/builder/src/third_party/dart/.packages --dfe=/b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/snapshots/kernel-service.dart.snapshot /b/s/w/ir/cache/builder/src/third_party/dart/pkg/front_end/tool/_fasta/compile_platform.dart --sound-null-safety --nnbd-agnostic --target=flutter_runner dart:core --single-root-scheme=org-dartlang-sdk --single-root-base=/b/s/w/ir/cache/builder/src/ org-dartlang-sdk:///flutter/shell/platform/fuchsia/flutter/kernel/libraries.json vm_outline_strong.dill flutter_runner_patched_sdk/platform_strong.dill flutter_runner_patched_sdk/vm_outline_strong.dill
Command failed: /b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/dart -Dsdk_hash=d2025958e3 --packages=/b/s/w/ir/cache/builder/src/third_party/dart/.packages --dfe=/b/s/w/ir/cache/builder/src/third_party/dart/tools/sdks/dart-sdk/bin/snapshots/kernel-service.dart.snapshot /b/s/w/ir/cache/builder/src/third_party/dart/pkg/front_end/tool/_fasta/compile_platform.dart --sound-null-safety --nnbd-agnostic --target=flutter_runner dart:core --single-root-scheme=org-dartlang-sdk --single-root-base=/b/s/w/ir/cache/builder/src/ org-dartlang-sdk:///flutter/shell/platform/fuchsia/flutter/kernel/libraries.json vm_outline_strong.dill flutter_runner_patched_sdk/platform_strong.dill flutter_runner_patched_sdk/vm_outline_strong.dill
output: Usage: compile_platform [options] dart-library-uri libraries.json vm_outline_strong.dill platform.dill outline.dill

Compiles Dart SDK platform to the Dill/Kernel IR format.

Frequently used options:

  -o <file> Generate the output into <file>.
  -h        Display this message (add -v for information about all options).
Error: Unknown option '--sound-null-safety'.
```

Reply:
(that's the fuchsia failure but the one here was the same)

Reply:
Oh gotcha, I forgot about that. Is this failing because we are passing in a .packages file instead of a package_config.json?

Reply:
It seems like Dart just doesn't like that argument there.

Reply:
I'm going to split this one line into a separate patch and land the rest.